crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1027,8 +1047,29 @@ where
                         self.try_make_sync_target_canonical();
                     }
                     BlockStatus::Disconnected { missing_parent } => {
-                        // continue downloading the missing parent
-                        self.sync.download_full_block(missing_parent.hash);
+                        // compare the missing parent with the canonical tip
+                        let canonical_tip_num = self.blockchain.canonical_tip().number;
+
+                        // if the number of missing blocks is greater than the max, run the
+                        // pipeline
+                        if missing_parent.number >= canonical_tip_num &&
+                            missing_parent.number - canonical_tip_num >
+                                self.pipeline_run_threshold
+                        {
+                            self.sync.set_pipeline_sync_target(missing_parent.hash)
```

Comment:
this is a bit dangerous imo, because this is not necessarily part of the chain
so I think we should use the finalized block of the latest FCU here

Reply:
switched to setting the finalized sync target here, but would like some input on the edge case where the latest finalized block is `None`:
https://github.com/paradigmxyz/reth/pull/3059/files#diff-6422221acf035d27429c58dd93438246d53b548efb00b9b3cfbd610c94f8c544R1059-R1064

Reply:
yeah, we can ignore this else branch because I don't think this is even reachable, since we only trigger pipeline on a FCU